### PR TITLE
Allow chains that can't introspect their own `ConsensusState` support IBC

### PR DESCRIPTION
**Pull request by seunlanlege**
ref https://github.com/cosmos/ibc/pull/839

### CODE REVIEW

File: proto/ibc/core/connection/v1/tx.proto [hunk 1]
```diff
@@ -67,6 +67,8 @@ message MsgConnectionOpenTry {
   ibc.core.client.v1.Height consensus_height = 11
       [(gogoproto.moretags) = "yaml:\"consensus_height\"", (gogoproto.nullable) = false];
   string signer = 12;
+  // extra data for state machaines that can't introspect their own consensus state.
```
Comment (crodriguezvega) [inline proto/ibc/core/connection/v1/tx.proto:70]:
```suggestion
  // extra data for state machines that can't introspect their own consensus state.
```

File: proto/ibc/core/connection/v1/tx.proto [hunk 2]
```diff
@@ -67,6 +67,8 @@ message MsgConnectionOpenTry {
...
+  bytes host_consensus_state_info = 13;
```
Comment (crodriguezvega) [inline proto/ibc/core/connection/v1/tx.proto:71]: To match the [spec changes that were merged](https://github.com/cosmos/ibc/pull/839)...

```suggestion
  bytes host_consensus_state_proof = 13;
```

File: proto/ibc/core/connection/v1/tx.proto [hunk 3]
```diff
@@ -94,6 +96,8 @@ message MsgConnectionOpenAck {
   ibc.core.client.v1.Height consensus_height = 9
       [(gogoproto.moretags) = "yaml:\"consensus_height\"", (gogoproto.nullable) = false];
   string signer = 10;
+    // extra data for state machaines that can't introspect their own consensus state.
+    bytes host_consensus_state_info = 11;
```
Comment (crodriguezvega) [inline proto/ibc/core/connection/v1/tx.proto:100]:
```suggestion
    bytes host_consensus_state_proof = 11;
```

**Review by AdityaSripal — APPROVED** [review]
Thanks @seunlanlege

**Comment by en** [comment]
Hi @seunlanlege and @AdityaSripal,
Why don't we define this field as optional? I think the cosmos-sdk based chain should not need this field.

**Pull request merged**